src/System.Runtime.Extensions/src/System.Runtime.Extensions.csproj

```diff
@@ -176,6 +176,35 @@
       <Link>Common\Interop\Windows\mincore\Interop.LookupAccountNameW.cs</Link>
     </Compile>
   </ItemGroup>
+  <ItemGroup Condition=" '$(TargetsWindows)' == 'true' And '$(CoreClrOrCorRt)' == 'true'"> <!-- Microsoft.Win32.Registry, needed for some Environment env var support -->
+    <Compile Include="..\..\Microsoft.Win32.Registry\src\Microsoft\Win32\Registry.cs" />
```

Comment:
I agree there isn't a clear winning option and I don't have any better ideas. I'm OK with option 4 I just wanted to get an idea of how much of a size impact it has, 20kb isn't the end of the world.